Engine/Animation/AnimController.h

```diff
@@ -40,7 +41,10 @@ class AnimController
 	void SetStateMachine(uint32_t state_machine_uuid);
 	void GetClipTransform(uint32_t skeleton_uuid, std::vector<math::float4x4>& pose);
 	void StartNextState(const std::string& trigger);
+	std::shared_ptr<Transition> active_transition;
```

Comment:
If you are making these one public just for the panelStateMachine, I think is better to put the state machine as friend class